app/jobs/restricted_device_password_emailing_for_setting_job.rb

```diff
@@ -0,0 +1,37 @@
+class RestrictedDevicePasswordEmailingForSettingJob < ApplicationJob
+  queue_as :default
+
+  def perform(setting_classname:, setting_id:)
+    @setting = setting_classname.constantize.find(setting_id)
```

Comment:
We could just use dependency injection here to pass the class instead of converting it to a String and then back again.
eg

````
      RestrictedDevicePasswordEmailingForSettingJob.perform_later(
        setting_id: setting.id,
        setting_class: setting.class
      )
````

and then...

````
@setting = setting_class.find(setting_id)
````

Reply:
Maybe ActiveJob accepts complex param types but I like to pass simple ones as recommended. 
For example, pass the user_id of an instance as param instead of the user itself.
I applied the same criteria here and passed just the name of the class instead of the class object itself?